app/src/main/java/com/willowtree/vocable/presets/Phrase.kt

```diff
@@ -1,26 +1,46 @@
 package com.willowtree.vocable.presets
 
+import android.content.res.Resources
 import android.os.Parcelable
 import com.willowtree.vocable.room.PhraseDto
 import com.willowtree.vocable.utils.locale.LocalesWithText
+import com.willowtree.vocable.utils.locale.text
 import kotlinx.parcelize.Parcelize
 
 sealed class Phrase : Parcelable {
     abstract val phraseId: Long
-    abstract val localizedUtterance: LocalesWithText?
     abstract val sortOrder: Int
+
+    abstract fun text(resources: Resources): String
 }
 
 @Parcelize
 data class CustomPhrase(
     override val phraseId: Long,
-    override val localizedUtterance: LocalesWithText?,
-    override val sortOrder: Int
-) : Phrase(), Parcelable
+    override val sortOrder: Int,
+    val localizedUtterance: LocalesWithText?,
+) : Phrase(), Parcelable {
+
+    override fun text(resources: Resources): String {
+        return localizedUtterance?.localizedText?.text() ?: ""
+    }
+}
+
+@Parcelize
+data class PresetPhrase(
+    override val phraseId: Long,
+    override val sortOrder: Int,
+    val utteranceStringRes: Int?,
```

Comment:
Since this is new code, can we make this non-null _now_?

Reply:
good call- updated!